src/Og.php

```diff
@@ -405,17 +403,25 @@ public static function getSelectionHandler(FieldDefinitionInterface $field_defin
     // @codingStandardsIgnoreEnd
     if (!\Drupal::service('og.group_audience_helper')->isGroupAudienceField($field_definition)) {
       $field_name = $field_definition->getName();
-      throw new \Exception("The field $field_name is not an audience field.");
+      throw new \Exception("The field $field_name is not a group audience field.");
     }
 
-    $default_options = [
+    $entity_type_id = $field_definition->getTargetEntityTypeId();
+    $definition = \Drupal::entityTypeManager()->getDefinition($entity_type_id);
+
+    $values = [];
+    if ($bundle_key = $definition->getKey('bundle')) {
+      $values[$bundle_key] = $field_definition->getTargetBundle();
+    }
+
+    $entity = \Drupal::entityTypeManager()->getStorage($entity_type_id)->create($values);
+
+    $options = [
       'target_type' => $field_definition->getFieldStorageDefinition()->getSetting('target_type'),
       'handler' => $field_definition->getSetting('handler'),
-      'field_mode' => 'default',
-    ] + $field_definition->getSetting('handler_settings');
-
-    // Override with passed $options.
-    $options = NestedArray::mergeDeep($default_options, $options);
+      'handler_settings' => $field_definition->getSetting('handler_settings'),
+      'entity' => $entity,
+    ];
```

Comment:
Providing settings under 'handler_settings' is deprecated in drupal:8.4.0 support for 'handler_settings' is removed from drupal:9.0.0. Move the settings in the root of the configuration array. See https://www.drupal.org/node/2870971 

```suggestion
      'entity' => $entity,
    ] + $field_definition->getSetting('handler_settings');
```